src/uint/div.rs

```diff
@@ -203,25 +216,65 @@ impl<const LIMBS: usize> Uint<LIMBS> {
 
     /// Perform checked division, returning a [`CtOption`] which `is_some`
     /// only if the rhs != 0
+    ///
+    /// ### Usage:
+    /// ```
+    /// use crypto_bigint::{U448, NonZero, subtle::{CtOption, Choice}};
+    ///
+    /// let a = U448::from(8_u64);
+    /// let result = NonZero::new(U448::from(4_u64))
+    ///     .map(|b| a.div_rem(&b))
+    ///     .expect("Division by zero");
+    ///
+    /// assert_eq!(result.0, U448::from(2_u64));
+    ///
+    /// // Check division by zero
+    /// let zero = U448::from(0_u64);
+    /// assert!(bool::from(a.checked_div(&zero).is_none()), "Should be None for division by zero");
+    /// ```
     pub fn checked_div(&self, rhs: &Self) -> CtOption<Self> {
         NonZero::new(*rhs).map(|rhs| {
             let (q, _r) = self.div_rem(&rhs);
             q
         })
     }
 
-    /// Wrapped (modular) remainder calculation is just `self` % `rhs`.
-    /// There’s no way wrapping could ever happen.
     /// This function exists, so that all operations are accounted for in the wrapping operations.
-    ///
     /// Panics if `rhs == 0`.
-    pub const fn wrapping_rem(&self, rhs: &Self) -> Self {
+    ///
+    /// ### Usage:
+    /// ```
+    /// use crypto_bigint::U448;
+    ///
+    /// let a = U448::from(10_u64);
+    /// let b = U448::from(3_u64);
+    /// let remainder = a.wrapping_rem_vartime(&b);
+    ///
+    /// assert_eq!(remainder, U448::from(1_u64));
+    /// ```
+    pub const fn wrapping_rem_vartime(&self, rhs: &Self) -> Self {
         let nz_rhs = rhs.to_nz().expect("non-zero divisor");
         self.rem_vartime(&nz_rhs)
     }
 
     /// Perform checked reduction, returning a [`CtOption`] which `is_some`
     /// only if the rhs != 0
+    ///
+    /// ### Usage:
+    /// ```
+    /// use crypto_bigint::{U448, NonZero, subtle::{Choice,CtOption}};
+    ///
+    /// let a = U448::from(10_u64);
+    /// let remainder_option = NonZero::new(U448::from(3_u64))
+    ///     .map(|b| a.rem(&b));
+    ///
+    /// assert!(bool::from(remainder_option.is_some()), "Reduction by zero");
```

Comment:
I'd find this more intuitive without the comment:

```suggestion
    /// assert!(bool::from(remainder_option.is_some()));
```